### PR TITLE
Added homepage and post features

### PR DESCRIPTION
**Pull request by parthgabani777**
- Added Homepage 
- Create/edit/delete post features
- Likes/dislike, bookmark add/remove feature

### CODE REVIEW

File: src/components/sidebar/left-sidebar.js [hunk 1]
```diff
@@ -22,6 +24,12 @@ export function LeftSidebar() {
         { url: "/setting", name: "Setting", icon: "fa-cog" },
     ];
 
+    // Add post modal toggler
+    const [showAddPostModal, setShowAddPostModal] = useState(false);
+    const showAddPostModalClickHandler = () => {
+        isAuthorized ? setShowAddPostModal(true) : navigation("/login");
```
Comment (ajay-mandaviya) [inline src/components/sidebar/left-sidebar.js:30]: just a small suggestion   directly navigate to login page better show alert Please Login to Create a Post

File: src/services/comment-services.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    } = await axios.post(
+        `/api/comments/add/${postId}`,
+        { commentData },
+        {
+            headers: { authorization: token },
+        }
+    );
+    return comments;
+};
+
```
Comment (ajay-mandaviya) [inline src/services/comment-services.js:16]: Don't ' return directly commetData response 
return data , it help to handle error , let's say you got  404 , 400 how you can manage it ?

File: src/slices/userSlice.js [hunk 3]
```diff
@@ -49,6 +49,22 @@ export const getUserPosts = createAsyncThunk(
...
+);
+
+export const removeBookmark = createAsyncThunk(
+    "posts/removeBookmark",
+    async ({ postId, token }) => {
+        const bookmarks = await removeBookmarkService(postId, token);
+        return bookmarks;
+    }
+);
+
```
Comment (ajay-mandaviya) [inline src/slices/userSlice.js:88]: also handle the error here like 

export const addBookMark = createAsyncThunk(
  "post/addBookMark",
  async (postId, thunkAPI) => {
    try {
      const token = localStorage.getItem("token");
      const respone = await addBookMarkPostApi(token, postId);

      return respone.data;
    } catch (error) {
      return thunkAPI.rejectWithValue(error.response.data);
    }
  }
);

**Review by ajay-mandaviya — APPROVED** [review]
Everything looks fine just add comment for better understanding 
you can merge it

File: src/pages/home/home.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    margin: auto;
+}
+.post-container {
+    display: flex;
+    flex-direction: column;
+    color: var(--light-white-color);
+    gap: 3rem;
+}
+.chips-container {
+    display: flex;
```
Comment (paarthbajaj) [inline src/pages/home/home.css:11]: You could make utility class for display: flex

**Review by paarthbajaj — APPROVED** [review]
The code looks great and well structured. I noticed that on click of like/heart on the home page, the page refreshes, but I expected the behavior should be similar to the click of heart icon on the profile page, i.e. page does not refresh. If possible, you can make this change, otherwise no worries. Well done, keep it up!

File: src/components/sidebar/left-sidebar.js (hunk 1 above)
Comment (parthgabani777) [reply src/components/sidebar/left-sidebar.js:30]: I will implement it in future pr. thank you for the suggestion.

**Pull request merged**